core/db/migrate/20141217215630_update_product_slug_index.rb

```diff
@@ -0,0 +1,6 @@
+class UpdateProductSlugIndex < ActiveRecord::Migration
+  def change
```

Comment:
I'd prefer to ship the "save" version by default. People with performance issues typically know that in advance and test / optimize migrations by hand before "blindly" applying the stock version. At least we do.